core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeStateListenerBase.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.metadata;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.NodeStateListener;
+
+/**
+ * Convenience class for node state listener implementations that only need to override one method.
+ */
+public class NodeStateListenerBase implements NodeStateListener {
```

Comment:
Should this and `SchemaChangeListenerBase` really be in the internal package?  It sort of makes sense given that it's not an interface, but I would also expect this to be used more often than not by users, and also wouldn't expect it to change, so it seems safe to be part of API, but was just wondering your thoughts on this.

Reply:
I was just wondering about that too. I think it makes sense to keep it in the API package after all.